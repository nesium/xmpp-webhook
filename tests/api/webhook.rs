use anyhow::Result;
use insta::assert_snapshot;

use crate::helpers::spawn_app;

#[tokio::test]
async fn test_push() -> Result<()> {
    let app = spawn_app().await;
    let client = reqwest::Client::new();

    let response = client
        .post(&format!("{}/webhook", &app.address))
        .header("X-GitHub-Event", "push")
        .header("Content-Type", "application/json")
        .body(include_str!("fixtures/push.json"))
        .send()
        .await?;

    assert!(response.status().is_success());
    assert_snapshot!(app.xmpp.sent_messages()[0].message);

    Ok(())
}

#[tokio::test]
async fn test_issue_opened() -> Result<()> {
    let app = spawn_app().await;
    let client = reqwest::Client::new();

    let response = client
        .post(&format!("{}/webhook", &app.address))
        .header("X-GitHub-Event", "issues")
        .header("Content-Type", "application/json")
        .body(include_str!("fixtures/issue_opened.json"))
        .send()
        .await?;

    assert!(response.status().is_success());
    assert_snapshot!(app.xmpp.sent_messages()[0].message);

    Ok(())
}

#[tokio::test]
async fn test_issue_comment_created() -> Result<()> {
    let app = spawn_app().await;
    let client = reqwest::Client::new();

    let response = client
        .post(&format!("{}/webhook", &app.address))
        .header("X-GitHub-Event", "issue_comment")
        .header("Content-Type", "application/json")
        .body(include_str!("fixtures/issue_comment_created.json"))
        .send()
        .await?;

    assert!(response.status().is_success());
    assert_snapshot!(app.xmpp.sent_messages()[0].message);

    Ok(())
}
