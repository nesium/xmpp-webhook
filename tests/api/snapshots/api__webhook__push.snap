---
source: tests/api/webhook.rs
expression: "sent_messages[0].message"
---
New commits pushed to [prose-core-client](https://github.com/prose-im/prose-core-client) on branch [vcard](https://github.com/prose-im/prose-core-client/tree/vcard)

```
refactor: Allow accessing ServerFeatures
```

**Commit**: [9cc3bc3](https://github.com/prose-im/prose-core-client/commit/9cc3bc379a1c8e1a2a99dbd1c9eef7095d90e7fc)  |  **Author**: [nesium](mailto:mb@nesium.com)

---

```
chore: Determine server support for vCard4 and PEP vCard avatar conversion
```

**Commit**: [fe1a899](https://github.com/prose-im/prose-core-client/commit/fe1a899f154654cbf16e566aa9a5e9a38027e94d)  |  **Author**: [nesium](mailto:mb@nesium.com)

---

```
feat: Introduce Avatar struct to support vCard and PEP avatars

refs #50, refs #78
```

**Commit**: [8b47ae3](https://github.com/prose-im/prose-core-client/commit/8b47ae324d2a156b2fd9103f9ccc3a23e5d8a666)  |  **Author**: [nesium](mailto:mb@nesium.com)

---

```
feat: Add client and caps on Participant
```

**Commit**: [7843210](https://github.com/prose-im/prose-core-client/commit/78432105260019f5ef33d3534ee8c099c9a34147)  |  **Author**: [nesium](mailto:mb@nesium.com)
